### PR TITLE
Category list

### PR DESCRIPTION
**Pull request by RemaaBdair**

### CODE REVIEW

File: src/pages/Category/components/CategoryGrid/CategoryGrid.stories.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
+import CategoryGrid from "./CategoryGrid";
+
+export default {
+  title: "CategoryPage",
+};
+
+export const categoryGrid = () => <CategoryGrid />;
```
Comment (Abdallatif) [inline src/pages/Category/components/CategoryGrid/CategoryGrid.stories.js:8]: this is not working in storybook

File: src/pages/Category/components/CategoriesList/CategoriesList.tsx [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  headerClassName: string,
+  NameAscendingOrder: boolean,
+  setNameAscendingOrder: React.Dispatch<React.SetStateAction<boolean>>,
+  DateAscendingOrder: boolean,
+  setDateAscendingOrder: React.Dispatch<React.SetStateAction<boolean>>,
+  setCategoryData: React.Dispatch<React.SetStateAction<Category[]>>
+) => {
+  return (
+    <>
+      <th key="name" className={headerClassName}>
```
Comment (Abdallatif) [inline src/pages/Category/components/CategoriesList/CategoriesList.tsx:56]: why aren't you using Table components from material-ui? https://material-ui.com/components/tables/#table

File: src/pages/Category/components/CategoriesList/CategoriesList.tsx [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+import { isArray } from "util";
+interface Props {
+  searchText: string;
+  entriesPerPage: number;
+  currentPageNumber: number;
+  handleNextPageEntries: (value: number) => void;
+  handleEntriesLength: (value: number) => void;
+}
+
+const getSortLabel = (
```
Comment (Abdallatif) [inline src/pages/Category/components/CategoriesList/CategoriesList.tsx:21]: this could be a component

File: src/pages/Category/components/CategoriesList/CategoriesList.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        )}
+      </th>
+      <th key="action" className={headerClassName}>
+        Action
+      </th>
+    </>
+  );
+};
+
+const getTableBody = (
```
Comment (Abdallatif) [inline src/pages/Category/components/CategoriesList/CategoriesList.tsx:81]: this could be a component

File: src/pages/Category/components/CategoriesList/CategoriesList.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+  const {
+    classes,
+    searchText,
+    entriesPerPage,
+    currentPageNumber,
+    handleNextPageEntries,
+    handleEntriesLength,
+  } = props;
+  const [categoryData, setCategoryData] = useState<Category[]>([]);
+  const [tableBody, setTableBody] = useState<JSX.Element | JSX.Element[]>([]);
```
Comment (Abdallatif) [inline src/pages/Category/components/CategoriesList/CategoriesList.tsx:145]: why is the table body a state?

File: src/pages/Category/Components/CategoriesList/util.tsx [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    method: "DELETE",
+  }).catch((error) => {
+    console.error("Error:", error);
+  });
+};
+export const sortTable = async (
+  orderBy: keyof Category,
+  ascOrder: boolean = true
+) => {
+  const categoriesData = await fetchCategories();
```
Comment (Abdallatif) [inline src/pages/Category/Components/CategoriesList/util.tsx:64]: do we want to fetch categories on each sort?

File: src/pages/Category/Components/DeleteCategoryDialog/DeleteCategoryDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+          type="submit"
+          variant="text"
+          fullWidth={false}
+          classes={{ root: classes.button }}
+        >
+          Cancel
+        </MyButton>
+        <MyButton
+          OnClickHandle={async () => {
+            await deleteCategory(id);
```
Comment (Abdallatif) [inline src/pages/Category/Components/DeleteCategoryDialog/DeleteCategoryDialog.tsx:49]: does this component have to worry about calling `deleteCategory` api? Maybe it's the parent responsibility 🤔

File: src/pages/Category/Components/CategoriesList/CustomizedTableBody.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+    page,
+    categoryData,
+    onOpenDialog,
+    searchText = "",
+    setRowsLength,
+  } = props;
+  let rows = categoryData.filter((category: Category) =>
+    category.name.toLocaleLowerCase().includes(searchText.toLocaleLowerCase())
+  );
+  setRowsLength(rows.length);
```
[the remark this answers is not in the document]
Comment (RemaaBdair) [reply src/pages/Category/Components/CategoriesList/CustomizedTableBody.tsx:45]: After I did this, sorting does not seem to work anymore. Shouldn't rows be changing as categoryData is changed?

[the remark this answers is not in the document]
Comment (Abdallatif) [reply src/pages/Category/Components/CategoriesList/CustomizedTableBody.tsx:45]: that's strange, why is the sorting related to `rowsLength`?
btw, you may want to move the filtering logic to the parent component, so `CustomizedTableBody` is only responsible for rendering the rows

File: tsconfig.json [hunk 9]
```diff
@@ -1,25 +1,19 @@
...
-      "esnext"
-    ],
+    "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
     "esModuleInterop": true,
     "allowSyntheticDefaultImports": true,
     "strict": true,
-    "forceConsistentCasingInFileNames": true,
+    "forceConsistentCasingInFileNames": false,
```
Comment (Abdallatif) [inline tsconfig.json:10]: why did you change this to false?

Comment (RemaaBdair) [reply tsconfig.json]: i was trying to solve naming folders issue so i changed it in case it will make a difference 😅

File: src/pages/Category/Components/CategoriesList/CustomizedTableBody.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (RemaaBdair) [reply src/pages/Category/Components/CategoriesList/CustomizedTableBody.tsx]: i moved the logic to parent component then switched the order of filteredData and sortedData and it works now. i have no idea why it worked.
